sweetercat/tests/test_app.py

```diff
@@ -167,6 +167,7 @@ def test_error_404(client):
     assert b'<img src="static/spacemonkey.png" alt="">' in error404.data
 
 
+@pytest.mark.xfail
```

Comment:
I fixed the problem with this in commit 0186a70a7.

The ordering of the null replacement is important because it changes the type of the columns.
Need to do the rounding first. Then the integers. Then the null replacement (changes column types which effects rounding and null detection). 
I also had to manually replace the nulls for temperature as it is turned into strings.
This was to stop nulls appearing as  `"."` or `nan` on the homepage.